src/prefect/core/flow.py

```diff
@@ -606,30 +606,55 @@ def chain(self, *tasks: Task, validate: bool = None) -> List[Edge]:
             )
         return edges
 
-    def update(self, flow: "Flow", validate: bool = None) -> None:
+    def update(
+        self, flow: "Flow", merge_parameters: bool = False, validate: bool = None
+    ) -> None:
         """
-        Take all tasks and edges in another flow and add it to this flow
+        Take all tasks and edges in another flow and add it to this flow.
+            When `merge_parameters` is set to`True` -- Duplicate parameters in the input `flow`
+            are replaced with those in the flow being updated.
 
         Args:
-            - flow (Flow): A flow which is used to update this flow
-            - validate (bool, optional): Whether or not to check the validity of the flow
+            - flow (Flow): A flow which is used to update this flow.
+            - merge_parameters (bool, False): Duplicate paramaeters in the input flow are replaced
+                with parameters from the updated flow. Validate defaults to `True`.
+            - validate (bool, optional): Whether or not to check the validity of the flow.
 
         Returns:
             - None
         """
+        if merge_parameters:
+            validate = True
+
         for task in flow.tasks:
-            if task not in self.tasks:
+
+            if merge_parameters and isinstance(task, Parameter):
+                duped_parameter = [p for p in self.parameters() if p.name == task.name]
+            else:
+                duped_parameter = []
+            # duped_task = self.get_tasks(slug=task.slug) if merge_parameters else list()
+
+            # Adds a task if slugs are not duplicate
+            if (task not in self.tasks) and (not duped_parameter):
                 self.add_task(task)
 
+        # Append the new edges to this flow.
         for edge in flow.edges:
-            if edge not in self.edges:
-                self.add_edge(
-                    upstream_task=edge.upstream_task,
-                    downstream_task=edge.downstream_task,
-                    key=edge.key,
-                    mapped=edge.mapped,
-                    validate=validate,
-                )
+
+            if edge.upstream_task.name in self.slugs.values():
+
+                upstream_task = self.get_tasks(name=edge.upstream_task.name)[0]
```

Comment:
within Prefect internals we need to be as precise as possible; using `name` to check against `slug` and then `get_tasks` with name could lead to all sorts of weird bugs.

I'm not 100% sure what the intent is here - it seems that this might be the place where the "merge" is happening, but it's not dependent on the value of `merge_parameters`?

Reply:
@cicdw Thank you for taking the time to review.

>  using name to check against slug and then get_tasks with name could lead to all sorts of weird bugs.

Yeah. Not ideal. It rests on two assumptions (a) Prefect does not change the slug naming convention for parameters and (b) Parameters will always be unique. But I think I can find some other way to compare slug vs slug.  Let me know if you have any suggestions.

> I'm not 100% sure what the intent is here - it seems that this might be the place where the "merge" is happening, but it's not dependent on the value of merge_parameters?

Correct.  It does not depend on `parameter_merge`. Because:  In the case of duplicate parameters and `parameter_merge` equal to `False` an error is raised before getting to edges,  If `merge_parameter` is `True`  than there are no parameter slug conflicts anymore .  But the parameter's downstream task needs to be replaced with the parameter of the flow being updated. #2723.  It's similar to `flow,replace` .

Reply:
Thanks for the context @grybox - I’ll get back to you with some thoughts

Reply:
Hey @gryBox - I did a quick pass on your branch (I hope that's OK!) and am going to merge -- thanks for the contribution!

Reply:
@cicdw  Thanks for all the help.  Can't wait to see if anybody else uses it.

Reply:
No problem! Marvin will stop greeting you on your PRs now that you have official "Contributor" status ;) 

I think @bnaul has expressed interest in this functionality before so he might be an early user